pcdsdevices/attenuator.py

```diff
@@ -441,3 +443,245 @@ class GasAttenuator(Device, BaseInterface):
+
+    last_energy = Cpt(
+        EpicsSignalRO, ':SYS:LastPhotonEnergy_RBV', kind='config',
+        doc=('The photon energy used for the previous calculation; i.e., '
+             'the one that goes along with `best_config`.'),
+    )
+
+    # NOTE: this variant exists as well but duplicates the bitmask information:
+    # best_config = Cpt(
+    #     EpicsSignalRO, ':SYS:BestConfiguration_RBV', kind='normal',
+    #     doc='The best configuration of filters for the desired transmission',
+    # )
+    # set_metadata(best_config, dict(variety='array-nd'))
+    # # TODO: array-tabular would be nice, but does not work in typhos yet
+
+    best_config_bitmask = Cpt(
+        EpicsSignalRO, ':SYS:BestConfigurationBitmask_RBV', kind='normal',
+        doc='The best configuration of filters for the desired transmission.',
+    )
+    set_metadata(best_config_bitmask, dict(variety='bitmask', bits=18))
+    # TODO: array-tabular would be nice, but does not work in typhos yet
+
+    best_config_error = Cpt(
+        EpicsSignalRO, ':SYS:BestConfigError_RBV', kind='normal',
+        doc='Desired to calculated transmission error',
+    )
+
+    # NOTE: this variant exists as well but duplicates the bitmask information:
+    # active_config = Cpt(
+    #     EpicsSignalRO, ':SYS:ActiveConfiguration_RBV', kind='omitted',
+    #     doc='Where the filters are now',
+    # )
+    # set_metadata(active_config, dict(variety='array-nd'))
+    # TODO: array-tabular would be nice, but does not work in typhos yet
+
+    active_config_bitmask = Cpt(
+        EpicsSignalRO, ':SYS:ActiveConfigurationBitmask_RBV', kind='normal',
+        doc='Where the filters are now (as an integer)',
+    )
+    set_metadata(active_config_bitmask, dict(variety='bitmask', bits=18))
+
+    # NOTE: this variant exists as well but duplicates the bitmask information:
+    # filters_moving = Cpt(
+    #     EpicsSignalRO, ':SYS:FiltersMoving_RBV', kind='normal',
+    #     doc='Filter-by-filter motion status (1 if moving)',
+    # )
+    # set_metadata(filters_moving, dict(variety='array-nd'))
+
+    filters_moving_bitmask = Cpt(
+        EpicsSignalRO, ':SYS:FiltersMovingBitmask_RBV', kind='normal',
+        doc='Filter-by-filter motion status as a bitmask',
+    )
+    set_metadata(filters_moving_bitmask, dict(variety='bitmask', bits=18))
+
+    run_calculation = Cpt(
+        EpicsSignal, ':SYS:Run', kind='config',
+        doc='Start the calculation',
+    )
+    set_metadata(run_calculation, dict(variety='command-proc', value=1))
+
+    apply_config = Cpt(
+        EpicsSignal, ':SYS:ApplyConfiguration', kind='config',
+        doc='Apply the best configuration (i.e., move the filters)',
+    )
+    set_metadata(apply_config, dict(variety='command-proc', value=1))
+
+    moving = Cpt(
+        EpicsSignal, ':SYS:Moving_RBV', kind='config',
+        doc='Are filters being moved in/out?',
+    )
+    set_metadata(moving, dict(variety='bitmask', bits=1))
```

Comment:
Does this give us a little light up box? If so it's a good trick I'd like to bring to other classes

Reply:
Yes:

<img width="539" alt="image" src="https://user-images.githubusercontent.com/5139267/89214328-7347cb00-d57b-11ea-8538-fffc2f378b4b.png">

(Although in pasting this, I realized I noticed that this should be `EpicsSignalRO` so it only generates the readback widget)

Reply:
Still beats `moving      1`